registry/coder/modules/agentapi/main.test.ts

```diff
@@ -148,4 +148,71 @@ describe("agentapi", async () => {
     ]);
     expect(respAgentAPI.exitCode).toBe(0);
   });
+
+  test("no-subdomain-base-path", async () => {
+    const { id } = await setup({
+      moduleVariables: {
+        agentapi_subdomain: "false",
+      },
+    });
+
+    const respModuleScript = await execModuleScript(id);
+    expect(respModuleScript.exitCode).toBe(0);
+
+    const agentApiProcessOutput = await execContainer(id, [
+      "bash",
+      "-c",
+      "ps -eo command | grep [a]gentapi",
+    ]);
+    expect(agentApiProcessOutput.exitCode).toBe(0);
+    expect(agentApiProcessOutput.stdout).toContain(
+      `--chat-base-path /@default/default.foo/apps/agentapi-web/chat`,
+    );
+  });
+
+  test("validate-agentapi-version", async () => {
+    const cases = [
+      {
+        moduleVariables: {
+          agentapi_version: "v0.3.2",
+          agentapi_subdomain: "false",
+        },
+        shouldThrow: "",
+      },
+      {
+        moduleVariables: {
+          agentapi_version: "v0.3.2",
+          agentapi_subdomain: "true",
+        },
+        shouldThrow: "",
+      },
+      {
+        moduleVariables: {
+          agentapi_version: "v0.3.1",
+          agentapi_subdomain: "false",
+        },
+        shouldThrow: "Running with subdomain = false is only supported by agentapi >= v0.3.2.",
+      },
+      {
+        moduleVariables: {
+          agentapi_version: "v0.3.1",
+          agentapi_subdomain: "true",
+        },
+        shouldThrow: "",
+      },
+      {
+        moduleVariables: {
+          agentapi_version: "arbitrary-string-bypasses-validation",
+        },
+        shouldThrow: "",
+      }
+    ];
+    for (const { moduleVariables, shouldThrow } of cases) {
+      if (shouldThrow) {
+        expect(setup({ moduleVariables: moduleVariables as Record<string, string> })).rejects.toThrow(shouldThrow);
+      } else {
+        expect(setup({ moduleVariables: moduleVariables as Record<string, string> })).resolves.toBeDefined();
```

Comment:
review: I initially had `await expect` but my linter advised me that this is unnecessary given that the test function is async.